x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/optimizer/Optimizer.java

```diff
@@ -1130,7 +1130,8 @@ protected LogicalPlan rule(LogicalPlan plan) {
 
             plan.forEachDown(a -> {
```

Comment:
@matriv I don't think this whole thing is still needed from 1131 till line 1141.

Reply:
This should kick in for queries that end up to an LocalExec query, so no actual ES query is executed but one row (with literals only) is returned.

Reply:
But probably there is no such case as for group by.
If you remove it nothing breaks?

Reply:
nope, all integration tests are fine, I still need to run the unit tests make sure all is ok. will let you know.

Reply:
Thank you!

Reply:
some unit tests fail. Example this: `testFoldingToLocalExecWithAggregationAndOrderByAndLimit`, only this `assertEquals(LocalExec.class, p.getClass());` as we removed it, but the result is good. I guess let's just keep it in

Reply:
Yes please. Could you check and revert those formatting changes I mentioned?

Reply:
yep sorry, that was intellij.